packages/flutter/lib/src/widgets/conduit.dart

```diff
@@ -0,0 +1,485 @@
+              },
+            ),
+          );
+        }
+      }
+      try {
+        _remoteChild = updateChild(_remoteChild, built, this);
+        //assert(_remoteChild != null);
+      } catch (e, stack) {
+        built = ErrorWidget.builder(
+          _debugReportException(
+            ErrorDescription('building remoteChild $widget'),
+            e,
+            stack,
+            informationCollector: () sync* {
+              yield DiagnosticsDebugCreator(DebugCreator(this));
+            },
+          ),
+        );
+        _remoteChild = updateChild(null, built, this);
+      }
+    });
+  }
+}
+
+/// A widget keyed by an [AnchorKey] and allows a [Conduit] possesses the same
+/// [AnchorKey] to attach the render object of its `remoteChild` to.
+class ConduitAnchor extends LeafRenderObjectWidget {
+  /// Creates a [ConduitAnchor] with an [AnchorKey].
+  const ConduitAnchor({
+    @required AnchorKey key,
+  }) : assert(key != null),
+       super(key: key);
+
+  @override
+  AnchorKey get key => super.key as AnchorKey;
+
+  @override
+  LeafRenderObjectElement createElement() => _ConduitAnchorElement(this);
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return _RenderConduitAnchor()..leader = key._conduitElement?.renderObject;
+  }
+}
+
+class _ConduitAnchorElement extends LeafRenderObjectElement {
+  _ConduitAnchorElement(ConduitAnchor widget) : super(widget);
+
+  @override
+  ConduitAnchor get widget => super.widget as ConduitAnchor;
+
+  @override
+  _RenderConduitAnchor get renderObject => super.renderObject as _RenderConduitAnchor;
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+    renderObject.leader = widget.key._conduitElement?.renderObject;
+  }
+
+  @override
+  void unmount() {
+    renderObject.leader = null;
+    super.unmount();
+  }
+
+  @override
+  void insertChildRenderObject(RenderBox child, dynamic slot) {
+    assert(slot == null);
+    final _RenderConduitAnchor renderBox = renderObject;
+    assert(renderBox.debugValidateChild(child));
+    renderObject.child = child;
+    assert(renderObject == renderBox);
+  }
+
+  @override
+  void removeChildRenderObject(RenderBox child) {
+    final _RenderConduitAnchor renderBox = renderObject;
+    // This is more forgiving than a typical removeChildRenderObject implementation,
+    // since _RenderConduitAnchorElement detaches its remoteChild.renderObject in
+    // `Element.deactivate()`.
+    if (renderBox.child != null) {
+      assert(renderBox.child == child);
+      renderBox.child = null;
+    }
+    assert(renderObject == renderBox);
+  }
+}
+
+// A RenderProxyBox subclass that keeps its follower's depth greater than or equal
+// to its own depth, if its follower is not null.
+class _RenderConduit extends RenderProxyBox {
+  _RenderConduit([RenderBox child]) : super(child);
+
+  _RenderConduitAnchor _follower;
```

Comment:
I was confused by the name. If i read it correctly, shouldn't the Conduit be follower and ConduitAnchor be the leader because conduit attach its child the conduitanchor?

Reply:
The anchor is named follower in the sense that it depends on its conduit. The anchor changes if its conduit changes, but not the other way around. I guess dependent/dependency sounds good too.

Reply:
The problem with naming the anchor follower is that you then have one follower with multiple leaders which is counterintuitive. Usually there should only be one leader with multiple followers. The dependent/dependency sounds good